src/main/java/com/puppycrawl/tools/checkstyle/utils/XpathUtil.java

```diff
@@ -0,0 +1,117 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2019 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.utils;
+
+import java.util.Arrays;
+import java.util.List;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Contains utility methods for xpath.
+ *
+ */
+public final class XpathUtil {
+
+    /**
+     * List of token types which support value attribute.
+     * These token types were selected based on analysis that all others do not match required
+     * criteria - value of the token must be useful and help to retrieve more precise results.
+     * There are three types of AST tokens:
+     * 1. Tokens for which the values are equal to the name of the token. Or in other words,
+     * nodes for which the following expression is always true:
+     * <pre>
+     *     detailAst.getText().equals(TokenUtil.getTokenName(detailAst.getType()))
+     * </pre>
+     * For example:
+     * <pre>
+     *     //MODIFIERS[@text='MODIFIERS']
+     *     //OBJBLOCK[@text='OBJBLOCK']
+     * </pre>
+     * These tokens do not match required criteria because their values do not carry any
+     * information, they do not affect the xpath requests and do not help to get more accurate
+     * results. The values of these nodes are useless. No matter which class you analyze, these
```

Comment:
"No matter what code"
there are not only classes in java.

Reply:
done